include/mamba/api/configuration.hpp

```diff
@@ -1443,6 +1466,11 @@ namespace mamba
             return p_impl->locked();
         }
 
+        bool is_valid_serialization(const std::string& value) const
```

Comment:
I'm not 100% sure of your question, is it related to type erasure ?
If so:
- we don't know what's the type of `Configurables` and need an interface to manipulate them
  -  we could also use dynamic polymorphism to pick the right implementation at runtime with a huge `if... else...` conditional approach
  - it's much more expressive to declare/add/register/insert a new `Configurable` in the `Configuration` singleton and use init value with type deduction
  - then just manipulate `Configurable` without its type using the type erased interface
- I guess some of forwarded calls could be improved at some point

Reply:
So the reason why we have to jump through so many hoops here is because of the type erasure?

I don't know if my question is related to type erasure. It's a broad question of why did you choose this way of structuring the code. I don't understand why we have a `Wrapper` class a `BaseWrapper` a diff. impl of the `is_valid_serialization` method for each one of them and an impl. for the `Configurable` class.
I was wondering if you're following some design model or if you just came up with this due the circumstances you had.